packages/swing-store/src/snapStore.js

```diff
@@ -203,7 +215,17 @@ export function makeSnapStore(
               compressedSize,
               compressedSnapshot,
             );
-
+            noteExport(
+              artifactName({ vatID, endPos }),
+              JSON.stringify({
+                vatID,
+                endPos,
```

Comment:
The "export data" record (in IAVL)'s value tells the recipient what to expect/require. That lets them compute an artifact filename to ask for, and also to validate the contents as it arrives. The artifact name is not the primary key.

I'll circle back to this once I've studied the PR more, but my first reaction is that artifact names should be distinct but not authoritative, and it's correct to put the critical validation data in the JSON blob rather than the artifact name.

Reply:
IMO the "entry" (key + value) in the "export data" forms the validation data, and the "key" part of it does not need to correspond 1:1 to artifact file names.

Reply:
We're already debating whether or not the endPos of a transcript span should be in the name or not.  I'd strongly prefer that changes to the naming conventions for an artifact not result in changes to the contents of the metadata record and vice-versa.

Reply:
I don't think naming convention for artifacts has anything to do with this observation. The metadata record is the composite of the key + value. The keys needs to be formed so that a simple lookup can be performed, aka keep the current form of `export.transcript.${vatID}.${startPos}. However the value does not need to repeat information already contained in the key, aka `vatID` and `startPos`. Repetition opens the door to question about what to do if the values diverge between key and value.

Reply:
I think this is a philosophical difference of opinion.  The information should not diverge; if it does, then there's a bug that we should learn about and suppressing information that would tell us about that bug is risky.